src/dosext/dpmi/dnative/dnative.c

```diff
@@ -85,9 +85,12 @@ static void copy_to_dpmi(sigcontext_t *scp, cpuctx_t *s)
   _C(trapno);
   _C(err);
   _C(cr2);
-  static_assert(sizeof(struct _fpstate) == sizeof(vm86_fpu_state),
-		"size mismatch");
-  scp->fpstate = (struct _fpstate *)&vm86_fpu_state;
+  if (scp->fpstate) {
+    // for i386, struct _fpstate also includes SSE fields
+    static_assert(sizeof(struct _fpstate) >= sizeof(vm86_fpu_state),
```

Comment:
This is somewhat nasty, i.e. the
size mismatch may mean that the
struct have really changed. So no
guarantee even the beginning matches.
How about instead add a needed
buffer into vm86_fpu_state under
ifdefs?
Also perhaps separating configure
change would be good. :)

Reply:
Also w/o adding a buffer, we do not
save/restore the full state that the signal
code in linux wants us to save/restore.
Who knows what problems will that bring.
What do you think about some hack, like
adding to the end of struct emu_fpstate
something like
`char buffer[sizeof(struct _libc_fpstate)-offsetof(struct emu_fpstate, buffer)];`?
Or for example revert to `aligned_alloc()`
but use the sizeof of struct _libc_fpstate?